assets/javascripts/discourse/components/campaign-banner.js.es6

```diff
@@ -0,0 +1,143 @@
+import { action } from "@ember/object";
+import Component from "@ember/component";
+import discourseComputed, { observes } from "discourse-common/utils/decorators";
+import { inject as service } from "@ember/service";
+import User from "discourse/models/user";
+import { Promise } from "rsvp";
+
+export default Component.extend({
+  router: service(),
+  dismissed: false,
+  loading: false,
+  showContributors: false,
+  contributors: [],
+  classNameBindings: [
+    "isSidebar:campaign-banner-sidebar",
+    "shouldShow:campaign-banner",
+  ],
+
+  init() {
+    this._super(...arguments);
+    const dismissed = document.cookie.includes(
+      "discourse-subscriptions-campaign-banner-dismissed"
+    );
+    this.set("dismissed", dismissed);
+
+    const contributorSetting = this.siteSettings
+      .discourse_subscriptions_campaign_contributors;
+
+    if (contributorSetting && contributorSetting !== "") {
+      this.set("loading", true);
+      let promises = [];
+      const contributorNames = [...new Set(contributorSetting.split(","))];
+
+      contributorNames.map((username) => {
```

Comment:
This is trivial but you could `let promises = ` this, since it's a map, and instead of pushing each promise you return the `promise` at the end of the map.